datatree/datatree.py

```diff
@@ -1412,3 +1413,8 @@ def to_zarr(
 
     def plot(self):
         raise NotImplementedError
+
+    def move_to(self, new_parent: Tree | None) -> DataTree:
+        self._set_parent(new_parent, self.name)
+        # would require us to complete some methods in the TreeNode Class for this to work
+        return Tree
```

Comment:
This should return `self` instead

Reply:
Yah, that makes sense. thanks @TomNicholas. I have also observed that the _set_parent ( ) method calls a _detach ( ) method. Inside the _detach ( ) method, it would make sense to delete this node from the children dictionary of the old parent that we are detaching from. we could say,
parent._children[self.name] = None

Reply:
@TomNicholas, some assistance please. i closed the pull request accidentally before effecting my new changes. is it possible for me to re-open it?